src/Sarif/Baseline/ResultMatching/SarifLogResultMatcher.cs

```diff
@@ -343,5 +347,24 @@ private List<ExtractedResult> ExtractResultsFromRuns(IEnumerable<Run> sarifRuns)
                 }
             }
         }
+
+        private static IList<T> MergeList<T>(IEnumerable<T> source1, IEnumerable<T> source2, IEqualityComparer<T> comparer)
+        {
+            var elementSet = new HashSet<T>(comparer);
+
+            List<T> combinedList = new List<T>();
+            combinedList.AddRange(source1 ?? Enumerable.Empty<T>());
```

Comment:
>?? Enumerable.Empty<T>() [](http://example.com/codeflow?start=42&length=24)

You could remove this since the caller already verifies it.

Reply:
I'd actually suggest removing the caller's handling, as it strengthens this method to have this code.

Reply:
1. I think if a helper method can handle null case inside it would be safer and may save some duplicated code for the person calling it.

2. another way is put different case of null check at the first few lines of the helper method which people don't like,
if source1 is not null and source2, do we even need to loop or just return source1

3. This method not just do merge but also do distinct, if source1 has 3 items and source2 is null, the result only have 2 items, the person call this method will be surprised. Should we call it like MergeAndDistinctList

(Just for discussion don't directly make changes according to my comment :) )

Reply:
refactor/simplified the method into an extension method, now it doesn't need to join the lists into 1 and null case handling is easier, pls see latest change.